script/cli.py

```diff
@@ -0,0 +1,214 @@
+
+        ----------------------------------------
+        test_photo_1.jpeg                Success
+        test_photo_2.jpeg                Success
+        test_photo_3.jpeg                Success
+        test_photo_4.jpeg                Success
+        test_photo_5.jpeg                Success
+        test_photo_6.jpeg                Success
+        test_photo_7.jpeg                Success
+        test_photo_8.jpeg                Success
+        test_photo_9.jpeg                Success
+        test_photo_10.jpeg               Success
+        test_photo_11.jpeg               Success
+        test_photo_12.jpeg               Success
+        ----------------------------------------
+        Done: 12 successful, 0 skipped, 0 failed
+
+    The above table might be generated as follows:
+
+        printer = ProgressPrinter()
+        successes = []
+        printer.line()
+        for i in range(1, 13):
+            printer.begin_status_line('test_photo_{}.jpeg'.format(i))
+            # Do work
+            successes.append(i)
+            printer.status_success()
+        printer.line()
+        printer.results(successes, [], [])
+
+    """
+
+    def __init__(self, quiet=False, width=40):
+        """Sets up new ProgressPrinter.
+
+        :param bool quiet: If True, this printer will never print anything.
+        :param int width: Width in characters of the table to print.
+        """
+        self.quiet = quiet
+        self.width = width
+
+    def results(self, successes, skips, failures):
+        """Print out the number of successes, skips, and failures that exist.
+        Output will be colored appropriately.
+
+        example::
+
+            printer.results(12, 0, 0)
+            # Done: 12 successful, 0 skipped, 0 failed
+
+        :param in successes: Number of successful operations.
+        :param in skips: Number of skipped operations.
+        :param in failures: Number of failed operations.
+        """
+        if self.quiet:
+            return
+
+        print "Done:",
+
+        if successes:
+            print CLIColor.ok_green("{} successful,".format(successes)),
+        else:
+            print "0 successful,",
+
+        print "{} skipped,".format(skips),
+
+        if failures:
+            print CLIColor.ok_green("{} failed".format(failures)),
+        else:
+            print "0 failed"
+
+    def line(self):
+        """Prints out a line of ``-`` characters to stdout."""
+        if not self.quiet:
+            print "-" * self.width
+
+    def begin_status_line(self, item):
+        """Print the first half of a status line to the console. After calling
+        this function, it is expected that a call to :func:`status_success`,
+        :func:`status_skip`, or :func:`status_failure` will follow.
+
+        :param str item: The text to print on the left half of the status line.
+        """
+        if not self.quiet:
+            print item + (" " * (self.width - 8 - len(item))),
```

Comment:
ahh, yes. It's 7 (the length of the string "success") plus 1 (python puts a single space character when you print two things. Example with width=20:

```python

item = "Filename.jpg"
print item + (" " * (30 - 8 - len(item))), # 10 spaces
print "Success"

#           11111111122222222223
# 123456789012345678901234567890
# ------------------------------
# Filename.jpg          <-- after line 2
# Filename.jpg           Success<-- after line 3
```